src/main/java/mate/academy/dao/BookDao.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.dao;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.repository.BookDaoRepository;
+import mate.academy.utility.Connector;
+
+@Dao
+public class BookDao implements BookDaoRepository {
+    private final Connector connector;
+
+    public BookDao() {
+        connector = new Connector();
+    }
```

Comment:
```suggestion
public class BookDao implements BookDaoRepository {
```
ConnectionUtil is a util class,  usually we don't have to create instances of that class